src/main/java/ua/tc/marketplace/config/SecurityConfig.java

```diff
@@ -3,33 +3,69 @@
 import jakarta.annotation.Nonnull;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
 import org.springframework.security.config.Customizer;
+import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.AbstractAuthenticationFilterConfigurer;
 import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
 import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.security.web.SecurityFilterChain;
 import org.springframework.web.servlet.config.annotation.CorsRegistry;
 import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+import ua.tc.marketplace.service.impl.UserDetailsServiceImpl;
 
 @EnableWebSecurity
 @Configuration
+@EnableMethodSecurity
 public class SecurityConfig {
-  private static final String[] WHITELIST = {"/api/v1/**", "/v3/api-docs/**",  "/swagger-ui/**"};
 
+  private static final String DEFAULT_SUCCESS_PAGE = "/api/v1/demo";
+  private static final String[] WHITELIST = {"/v3/api-docs/**", "/swagger-ui/**", DEFAULT_SUCCESS_PAGE};
+  private static final String CREATE_USER_POST_URL = "/api/v1/user";
+
+  @Bean
+  public UserDetailsService userDetailsService() {
+    return new UserDetailsServiceImpl();
+  }
 
   @Bean
   SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
     http
         .csrf(AbstractHttpConfigurer::disable)
         .cors(Customizer.withDefaults())
-        .sessionManagement(config -> config.sessionCreationPolicy(SessionCreationPolicy.STATELESS))
+//        .sessionManagement(config -> config.sessionCreationPolicy(SessionCreationPolicy.STATELESS))
```

Comment:
why did you comment this line? As i know RESTful apps should have stateless sessions, it means that every request should contain jwt token

Reply:
We have session authentification. Even though I also wanted jwt, but the examples i've found for that moment adviced implementing  tons of additional classes to handle JWT tokens. And I thought it was too big job for one ticket and went with easier session managment. It required commenting this line. Now I did some deeper search and found good example of Spring Security project with jwt authentification.
[https://medium.com/@tericcabrel/implement-jwt-authentication-in-a-spring-boot-3-application-5839e4fd8fac]
 But let's leave it for another ticket